### PR TITLE
Fix: Raised error in case file name is too long on Windows

### LINKED ISSUE #79: Save to Zarr fails on Windows if filename exceeds 256 characters
**Describe the bug**
Windows has a built-in maximum length of file names of 256 (see e.g. [here](https://it.cornell.edu/shared-file/windows-file-name-or-destination-path-you-specified-not-valid-or-too-long#:~:text=Microsoft%20Windows%20has%20a%20MAX_PATH,%2Frename%20these%20paths%2Ffiles.)). When Zarr attempts to create files, this can throw a cryptic `FileNotFoundError`.

**To Reproduce**
This simple code shows the faulty behavior:
```
import zarr

# simple function to create a dataset in a group
def create_zarr_group_dataset(root_path):
    if Path(root_path).is_dir():
        shutil.rmtree(root_path)
    zarr_root = zarr.open(root_path, mode="w", storage_options=None)
    zarr_group = zarr_root.create_group("a_group")
    g_dset = zarr_group.create_dataset(name="group_data", data=[str(i) for i in range(100)], 
                                       compressor=None)

# here we extend the file name by appending \\new_folder n_iter times
for n_iter in range(20):
    zarr_path = base_folder
    for i in range(n_iter):
        zarr_path += "\\new_folder"
    zarr_path += zarr_name
    print(f"N iter: {n_iter} - len file path {len(zarr_path)}")
    try:
        create_zarr_group_dataset(zarr_path)
    except Exception as e:
        print(f"Failed for iter {n_iter}")
```

Which produces:
```
N iter: 0 - len file path 57
N iter: 1 - len file path 68
N iter: 2 - len file path 79
N iter: 3 - len file path 90
N iter: 4 - len file path 101
N iter: 5 - len file path 112
N iter: 6 - len file path 123
N iter: 7 - len file path 134
N iter: 8 - len file path 145
N iter: 9 - len file path 156
N iter: 10 - len file path 167
N iter: 11 - len file path 178
N iter: 12 - len file path 189
N iter: 13 - len file path 200
Failed for iter 13
N iter: 14 - len file path 211
Failed for iter 14
N iter: 15 - len file path 222
Failed for iter 15
N iter: 16 - len file path 233
Failed for iter 16
N iter: 17 - len file path 244
Failed for iter 17
N iter: 18 - len file path 255
Failed for iter 18
N iter: 19 - len file path 266
Failed for iter 19
```

**Expected behavior**
We should raise an error with an informative message to reduce the **depth** of the destination folder.
Also opened an issue on the zarr project: https://github.com/zarr-developers/zarr-python/issues/1235

**Desktop (please complete the following information):**
 - OS: Windows

### PR DESCRIPTION
**Pull request by alejoe91**
Closes #79

### CODE REVIEW

**Comment by alejoe91** [comment]
Note that Zarr adds many characters for when creating groups, datasets, and temporary files, so I set a MAX_FILENAME_LEN of 100 in the script.

**Comment by jtyoung84** [comment]
I bumped the version. I need to figure out how to re-run the checks if another PR gets merged into main.

File: src/aind_data_transfer/writers.py [hunk 1]
```diff
@@ -38,6 +43,13 @@ def compress_and_write_block(
             experiment_name = read_block["experiment_name"]
             stream_name = read_block["stream_name"]
             zarr_path = output_dir / f"{experiment_name}_{stream_name}.zarr"
+            print(len(str(zarr_path)))
```
Comment (jtyoung84) [inline src/aind_data_transfer/writers.py:46]: Can we replace the print statement with a logging.debug message?

Comment (alejoe91) [reply src/aind_data_transfer/writers.py:46]: Oh that's a leftover print. Will remove it!

**Review by jtyoung84 — APPROVED** [review]

**Pull request merged**